File: Java/src/3Sum Closest .java
public class Solution {
    public int threeSumClosest(int[] num, int target) {
        Arrays.sort(num);
        int ret = num[0]+num[1]+num[num.length -1];
        for (int i = 0; i<num.length-2; i++){
            int left = i+1;
            int right = num.length - 1;
            while(left < right){
                int sum = num[i] + num[left] + num[right];
                if (sum > target){
                    right-- ;
                }
                else if(sum< target){
                    left++;
                }
                else{
                    return sum;
                }

                if (Math.abs(sum-target) < Math.abs(ret-target)){
                    ret = sum;
                }
            }
        }
        return ret;
    }
}

File: Python/3Sum Closest.py
class Solution:
    # @return an integer
    def threeSumClosest(self, num, target):
        if len(num) <= 3:
            return sum(num)
        
        num.sort()
        ret = num[0] + num[1] + num[2]
        for i in range(len(num) -2):
            j = i + 1
            k = len(num) -1
            while j<k :
                accurate = num[i] + num[j] + num[k]
                if abs(target - ret) > abs(target - accurate):
                    ret = accurate
                    if ret == target:
                        return ret
                if accurate > target:
                    k = k - 1
                else:
                    j = j + 1
                
        return ret
        
        